src/search_buffer.rs

```diff
@@ -118,39 +129,53 @@ impl<'a, W: WriteColor> BufferSearcher<'a, W> {
             return 0;
         }
 
-        self.match_count = 0;
+        self.matching_line_count = 0;
         self.line_count = if self.opts.line_number { Some(0) } else { None };
+        self.match_count = if self.opts.count_matches { Some(0) } else { None };
         let mut last_end = 0;
         for m in self.grep.iter(self.buf) {
             if self.opts.invert_match {
                 self.print_inverted_matches(last_end, m.start());
             } else {
                 self.print_match(m.start(), m.end());
+                self.count_individual_matches(m.start(), m.end());
             }
             last_end = m.end();
-            if self.opts.terminate(self.match_count) {
+            if self.opts.terminate(self.matching_line_count) {
                 break;
             }
         }
-        if self.opts.invert_match && !self.opts.terminate(self.match_count) {
+        if self.opts.invert_match && !self.opts.terminate(self.matching_line_count) {
             let upto = self.buf.len();
             self.print_inverted_matches(last_end, upto);
         }
-        if self.opts.count && self.match_count > 0 {
-            self.printer.path_count(self.path, self.match_count);
+        if self.opts.count && self.matching_line_count > 0 {
+            self.printer.path_count(self.path, self.matching_line_count);
+        } else if self.opts.count_matches
+            && self.match_count.map_or(false, |c| c > 0) {
```

Comment:
When wrapping a conditional to multiple lines, could you move the brace down to the next line so that the division between the conditional and the conditional body is more clear?